### ISSUE TITLE
Kubernetes 1.20 and selfLink usage in Packet CSI

### ISSUE THREAD

**Issue by displague** [packethost/csi-packet#99]
From @yp28 via Slack: https://eqix-metal-community.slack.com/archives/C71NJL1MH/p1610638537001600

It seems that packet-csi does not work by default anymore on Kubernetes 1.20 due to `selfLink` being deprecated and removed from default configuration now.

```
I0114 15:27:13.537726       1 controller.go:926] provision "default/podpvc" class "csi-packet-standard": started
E0114 15:27:13.547119       1 controller.go:943] provision "default/podpvc" class "csi-packet-standard": unexpected error getting claim reference: selfLink was empty, can't make reference
```

**Comment by displague** [comment]
This stems from https://github.com/kubernetes/enhancements/issues/1164

There are no references to selfLink in this repo, I suspect the problem here is with the version of external-provisioner we are using. 1.0.1 is well behind the current release: https://github.com/kubernetes-csi/external-provisioner/releases (the image URL has changed too).

It seems we will need at least 1.4.0 of external-provisioner to get 1.20+ support: https://kubernetes-csi.github.io/docs/external-provisioner.html.

**Comment by displague** [comment]
@yp28 I don't see many breaking changes between versions.

Updating the csi-provisioner image to:
```
image: quay.io/k8scsi/csi-provisioner:v1.5.0
```
and removing the `- "---provisioner=net.packet.csi"` argument (which became deprecated and ignored) may get things working again.

We should investigate moving to an even newer release, v2.x in `deploy/kubernetes/controller.yaml` to get ahead of this issue.

**Comment by yp28** [comment]
@displague switching to `csi-provisioner:v1.5.0` seems to work, although you will need to switch the `provisioner` field in the `StorageClass` around as well, from `csi.packet.net` to `net.packet.csi`.

**Comment by yp28** [comment]
There is also an additional switch to a newer version of the `csi-attacher` needed, due to the switch from `CSINodeInfo` to `CSINode` ([info](https://kubernetes-csi.github.io/docs/csi-node-object.html#crd-to-built-in-type)). I'm testing with `quay.io/k8scsi/csi-attacher:v2.2.0` and for now it seems to work.

**Comment by displague** [comment]
Will not fix - see #103 (Equinix Metal Block Storage from Datera is EOL).
